Source/Core/Core/HW/EXI/EXI_DeviceModem.cpp

```diff
@@ -0,0 +1,398 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/HW/EXI/EXI_DeviceModem.h"
+
+#include <inttypes.h>
+
+#include <algorithm>
+#include <cstring>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "Common/BitUtils.h"
+#include "Common/ChunkFile.h"
+#include "Common/CommonTypes.h"
+#include "Common/Logging/Log.h"
+#include "Common/Network.h"
+#include "Common/StringUtil.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/CoreTiming.h"
+#include "Core/HW/EXI/EXI.h"
+#include "Core/HW/Memmap.h"
+#include "Core/PowerPC/PowerPC.h"
+#include "Core/System.h"
+
+namespace ExpansionInterface
+{
+
+CEXIModem::CEXIModem(Core::System& system, ModemDeviceType type) : IEXIDevice(system)
+{
+  switch (type)
+  {
+  case ModemDeviceType::TAPSERVER:
+    m_network_interface = std::make_unique<TAPServerNetworkInterface>(
+        this, Config::Get(Config::MAIN_MODEM_TAPSERVER_DESTINATION));
+    INFO_LOG_FMT(SP1, "Created tapserver physical network interface.");
+    break;
+  }
+
+  m_regs[Register::DEVICE_TYPE] = 0x02;
+  m_regs[Register::INTERRUPT_MASK] = 0x02;
+}
+
+CEXIModem::~CEXIModem()
+{
+  m_network_interface->RecvStop();
+  m_network_interface->Deactivate();
+}
+
+bool CEXIModem::IsPresent() const
+{
+  return true;
+}
+
+void CEXIModem::SetCS(int cs)
+{
+  m_transfer_descriptor = INVALID_TRANSFER_DESCRIPTOR;
+}
```

Comment:
Shouldn't this only apply if `cs` is true, similar to this?

https://github.com/dolphin-emu/dolphin/blob/a5e410df11b83646f21941caba98674d807daa7e/Source/Core/Core/HW/EXI/EXI_DeviceEthernet.cpp#L106-L113

(I noticed this with b5a6fc11bc362da7d7b62a352a62dca9922f5963 on #12680, as there I changed how `SetCS` works and used `if (!was_selected && is_selected)` for setting it to invalid. But I'm not 100% sure if that actually makes sense or not.)